app/src/main/java/com/example/pehchaan/fragment/EmergencyFragment.java

```diff
@@ -244,14 +266,22 @@ public void onClick(View view) {
             @Override
             public void onClick(View view) {
 
-                // save all the contact number in the EmergencyContactNumber array
-                EmergencyContactNumber.add(num1.getString("num1", "Enter Num1"));
-                EmergencyContactNumber.add(num2.getString("num2", "Enter Num2"));
-                EmergencyContactNumber.add(num3.getString("num3", "Enter Num3"));
-                EmergencyContactNumber.add(num4.getString("num4", "Enter Num4"));
-
+                if (EmergencyContactNumber.isEmpty()) {
+                    EmergencyContactNumber.add(num1.getString("num1", "Enter Num1"));
+                    EmergencyContactNumber.add(num2.getString("num1", "Enter Num2"));
+                    EmergencyContactNumber.add(num3.getString("num1", "Enter Num3"));
+                    EmergencyContactNumber.add(num4.getString("num1", "Enter Num4"));
+                } else {
+                    EmergencyContactNumber.set(0, num1.getString("num1", "Enter Num1"));
+                    EmergencyContactNumber.set(1, num2.getString("num2", "Enter Num2"));
+                    EmergencyContactNumber.set(2, num3.getString("num3", "Enter Num3"));
+                    EmergencyContactNumber.set(3, num4.getString("num4", "Enter Num4"));
+                }
                 CustomEmergencyMess = nummsg.getString("nummsg", "Enter your message");
 
+                locationRequest();
+
+
                 /*Intent sms1 = new Intent(Intent.ACTION_VIEW);
                 sms1.setData(Uri.parse("smsto:"));
```

Comment:
do we need this commented code?

Reply:
Actually, it was already present; that's why I didn't remove it. But it is not necessary.